src/bit-systems/network-receive-system.ts

```diff
@@ -1,23 +1,49 @@
 import { addComponent, defineQuery, enterQuery, hasComponent, removeComponent, removeEntity } from "bitecs";
 import { HubsWorld } from "../app";
 import { Networked, Owned } from "../bit-components";
-import { createNetworkedEntityFromRemote } from "../utils/create-networked-entity";
-import { networkableComponents, schemas } from "../utils/network-schemas";
-import type { StringID, UpdateMessage } from "../utils/networking-types";
+import { renderAsNetworkedEntity } from "../utils/create-networked-entity";
+import { networkableComponents, schemas, StoredComponent } from "../utils/network-schemas";
+import type { ClientID, CursorBufferUpdateMessage, EntityID, StringID, UpdateMessage } from "../utils/networking-types";
 import { hasPermissionToSpawn } from "../utils/permissions";
+import { tryUnpin } from "../utils/store-networked-state";
 import { takeOwnershipWithTime } from "../utils/take-ownership-with-time";
 import {
   createMessageDatas,
-  isNetworkInstantiated,
+  isPinned,
   localClientID,
   networkedQuery,
   pendingMessages,
-  pendingParts
+  pendingParts,
+  softRemovedEntities
 } from "./networking";
 
+function isCursorBufferUpdateMessage(update: any): update is CursorBufferUpdateMessage {
+  return !!update.hasOwnProperty("componentIds");
+}
+
+function breakTie(a: ClientID, b: ClientID) {
+  if (a === "reticulum") return b;
+  if (b === "reticulum") return a;
+  // arbitrary (but consistent) tiebreak
+  return a < b ? a : b;
+}
+
+function isOutdatedMessage(eid: EntityID, updateMessage: UpdateMessage) {
+  const entityHasOwner = !!Networked.owner[eid];
+  const messageIsNewer = Networked.lastOwnerTime[eid] < updateMessage.lastOwnerTime;
```

Comment:
> It's not the message that is newer per se, but the owner.

True
> Might read more clearly as `ownerIsNewer`

There are two `owner`s (and `lastOwnerTime`s), so a name like `ownerIsNewer` would have me ask "_Which_ owner is newer?"

I think you would not have named this bool at all so I feel responsible for coming up with a better name... `messageOwnerIsNewer`? 

I think my favorite implementation of this function is still this one, even though `if ( predicate ) return bool` is a code smell and this can be expressed with ANDs and ORs:

```typescript
function isOutdatedMessage(eid: EntityID, updateMessage: UpdateMessage) {
  if (!Networked.owner[eid]) return false;
  if (Networked.lastOwnerTime[eid] < updateMessage.lastOwnerTime) return false;
  if (Networked.lastOwnerTime[eid] > updateMessage.lastOwnerTime) return true;
  return updateMessage.owner !== breakTie(APP.getString(Networked.owner[eid])!, updateMessage.owner);
}
```